wal/wal.go

```diff
@@ -195,6 +195,7 @@ func Create(lg *zap.Logger, dirpath string, metadata []byte) (*WAL, error) {
 				zap.Error(perr),
 			)
 		}
+		w.cleanupWAL(lg)
```

Comment:
add a defer func to do this?

```go
defer func() {
    if err!= nil {
        w.cleanupWAL(lg)
    }
}
```

Reply:
then we do not need to repeat the cleanup 3 times.

Reply:
Nice!